lib/internal/http2/compat.js

```diff
@@ -387,9 +477,16 @@ class Http2ServerResponse extends Stream {
     if (typeof name !== 'string')
       throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'name', 'string');
 
+    if (this[kStream].headersSent)
+      throw new errors.Error('ERR_HTTP2_HEADERS_SENT');
+
```

Comment:
We check that `name` is valid but not `value`?

Reply:
`assertValidHeader(name, value);` takes care of value. We're going to properly validate name & value further downstream at some point, not yet implemented.